File: Android/Wing/src/main/java/fun/chaoxi/wing/util/AppUtil.java
package fun.chaoxi.wing.util;

/**
 * @author Wing Gao
 * @date 2019/1/8 15:27
 * @description
 */
public class AppUtil {

    /**
     * 判断字符串是否有内容
     * @param str 被判断的字符串
     * @return 是否有内容
     */
    public static boolean isEmpty(String str) {
        return str == null || str.isEmpty() || "Null".equalsIgnoreCase(str);
    }
}

File: Android/Wing/src/main/java/fun/chaoxi/wing/util/AppUtil.kt
package `fun`.chaoxi.wing.util

import android.content.Context
import android.location.LocationManager


/**
 * @author Wing Gao
 * @date 2019/1/4 10:13
 * @description
 */

/**
 * 判断GPS是否可用
 */
fun isGpsEnable(context: Context): Boolean {
    val locationManager  = context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
    val gps = locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)
    val network = locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER)
    return gps || network
}